src/main/java/com/couchbase/lite/replicator/ChangeTracker.java

```diff
@@ -630,10 +644,11 @@ public void setPaused(boolean paused) {
 
     protected void waitIfPaused(){
         synchronized (pausedObj) {
-            while (paused) {
+            while (paused && running) {
                 Log.v(Log.TAG, "Waiting: " + paused);
                 try {
-                    pausedObj.wait();
+                    // every 30 sec, wake by myself to check if still needs to pause
+                    pausedObj.wait(TIMEOUT_FOR_PAUSE);
```

Comment:
Not sure 30 seconds is too long or not. Bringing to down to ~5 seconds might be a good idea (shouldn't cost performance impact).